### PR TITLE
Updated for Android 15, API Level 35 and Java 17

### PR DESCRIPTION
**Pull request by MahabubKarim**

### CODE REVIEW

File: .github/FUNDING.yml
Comment (vipulasri) [inline .github/FUNDING.yml:1]: I don't think these are necessary changes

File: app/build.gradle [hunk 1]
```diff
@@ -1,64 +1,69 @@
...
     implementation fileTree(dir: 'libs', include: ['*.jar'])
-    androidTestImplementation 'androidx.test:runner:1.2.0'
-    androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
+    androidTestImplementation 'androidx.test:runner:1.6.2'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.6.1'
+    implementation 'com.google.android.material:material:1.12.0'
 
     //kotlin
-    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:${versions.kotlin.stdlib}"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.9.10"
```
Comment (vipulasri) [inline app/build.gradle:48]: Avoid using hardcoded versions, better to use the version catalog for every dependency

File: app/build.gradle [hunk 2]
```diff
@@ -1,64 +1,69 @@
...
-    implementation "androidx.appcompat:appcompat:${versions.androidx.appCompat}"
-    implementation "androidx.constraintlayout:constraintlayout:${versions.androidx.constraintLayout}"
-    implementation "com.google.android.material:material:${versions.android.material}"
+    implementation "androidx.appcompat:appcompat:1.7.0"
+    implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
+    implementation "com.google.android.material:material:1.12.0"
 
     //misc
-    implementation 'com.dmitrymalkovich.android:material-design-dimens:1.4'
+    implementation 'com.dmitrymalkovich.android:material-design-dimens:1.2'
```
Comment (vipulasri) [inline app/build.gradle:58]: Why did we downgrade the library?

File: app/build.gradle [hunk 3]
```diff
@@ -1,64 +1,69 @@
...
     }
-    implementation ('org.adw.library:discrete-seekbar:1.0.1') {
-        exclude group: 'com.android.support'
-        exclude module: 'appcompat-v7'
-        exclude module: 'support-v4'
-    }
     implementation 'com.jakewharton.threetenabp:threetenabp:1.2.1'
     implementation project (':timelineview')
-    //implementation 'com.github.vipulasri:timelineview:1.1.0'
+    implementation project (':discreteseekbar')
```
Comment (vipulasri) [inline app/build.gradle:66]: Why are we adding whole library project as dependency?

File: timelineview/build.gradle [hunk 4]
```diff
@@ -1,35 +1,15 @@
-apply plugin: 'com.android.library'
-
-ext {
```
Comment (vipulasri) [inline timelineview/build.gradle:3]: This configuration is absolutely necessary for library deployment

File: timelineview/build.gradle [hunk 5]
```diff
@@ -1,35 +1,15 @@
...
-    buildToolsVersion versions.buildTools
+    namespace "com.github.vipulasri.timelineview"
+    compileSdk 35
+
     defaultConfig {
-        minSdkVersion versions.minSdk
-        targetSdkVersion versions.targetSdk
-        versionCode 12
-        versionName libraryVersion
+        minSdk 21
```
Comment (vipulasri) [inline timelineview/build.gradle:10]: We should use a common version configuration instead of hard-coded it in library and sample.

**Comment by vipulasri** [comment]
Thank you for your PR, and suggestions but these changes are not adding value to the project.

File: app/build.gradle (hunk 3 above)
Comment (MahabubKarim) [reply app/build.gradle:66]: Because discreteSeekBar is not working on your existing project.

File: app/build.gradle (hunk 2 above)
Comment (MahabubKarim) [reply app/build.gradle:58]: Okay I will change this.

File: timelineview/build.gradle (hunk 4 above)
Comment (MahabubKarim) [reply timelineview/build.gradle:3]: ok

**Pull request closed without merging**